plugins/orderConstraintsFilter.go

```diff
@@ -34,51 +35,50 @@ func MakeFilterOrderConstraints(
 
 // Apply impl.
 func (f *orderConstraintsFilter) Apply(
-	ops []build.TransactionMutator,
+	ops []txnbuild.Operation,
 	sellingOffers []hProtocol.Offer,
 	buyingOffers []hProtocol.Offer,
-) ([]build.TransactionMutator, error) {
+) ([]txnbuild.Operation, error) {
 	numKeep := 0
 	numDropped := 0
-	filteredOps := []build.TransactionMutator{}
+	filteredOps := []txnbuild.Operation{}
 
 	for _, op := range ops {
 		var keep bool
 		var e error
-		var opPtr *build.ManageOfferBuilder
+		var opPtr *txnbuild.ManageSellOffer
 
 		switch o := op.(type) {
-		case *build.ManageOfferBuilder:
+		case *txnbuild.ManageSellOffer:
 			keep, e = f.shouldKeepOffer(o)
 			if e != nil {
 				return nil, fmt.Errorf("could not transform offer (pointer case): %s", e)
 			}
 			opPtr = o
-		case build.ManageOfferBuilder:
-			keep, e = f.shouldKeepOffer(&o)
-			if e != nil {
-				return nil, fmt.Errorf("could not check transform offer (non-pointer case): %s", e)
-			}
-			opPtr = &o
 		default:
 			keep = true
 		}
 
+		amountFloat, e := strconv.ParseFloat(opPtr.Amount, 64)
+		if e != nil {
+			return nil, fmt.Errorf("could not convert amount to float: %s", e)
+		}
+
 		if keep {
 			filteredOps = append(filteredOps, opPtr)
 			numKeep++
 		} else {
 			numDropped++
 			// figure out how to convert the offer to a dropped state
-			if opPtr.MO.OfferId == 0 {
+			if opPtr.OfferID == 0 {
 				// new offers can be dropped, so don't add to filteredOps
-			} else if opPtr.MO.Amount != 0 {
+			} else if amountFloat != 0 {
 				// modify offers should be converted to delete offers
 				opCopy := *opPtr
-				opCopy.MO.Amount = 0
-				filteredOps = append(filteredOps, opCopy)
+				opCopy.Amount = "0"
+				filteredOps = append(filteredOps, &opCopy)
```

Comment:
why do we use a pointer of `opCopy` here whereas on the left hand side of this diff we are passing the underlying struct?

Reply:
`filteredOps` is an array of `txnbuild.Operation` and it can only be pointers. Same behaviour as `mo` above